js/client.js

```diff
@@ -1011,6 +1021,72 @@
 				break;
 			}
 		},
+		parseGroups: function (groupsList) {
+			var data = null;
+			try {
+				data = JSON.parse(groupsList);
+			} catch (e) {}
+			if (!data) return; // broken JSON - keep default ranks
+
+			var authTiers = [[], [], []]; // we will be splitting them into three tiers, before compiling them back into the main tree
```

Comment:
```
|groups|[{"symbol":">","name":"Meme","type":2},{"symbol":"~","name":"Administrator","type":2},{"symbol":"&","name":"Leader","type":2},{"symbol":"#","name":"Room Owner","type":2},{"symbol":"☆","name":"Player","type":0},{"symbol":"*","name":"Bot","type":0},{"symbol":"@","name":"Moderator","type":1},{"symbol":"%","name":"Driver","type":1},{"symbol":"|","name":"Operator","type":0},{"symbol":"+","name":"Voice","type":0},{"symbol":" ","name":null,"type":0}]
```
The server sends out this order in terms of the ranks (this is the hierarchy of jurisdiction for PS from greatest to least) along with their properties (2: upper staff - root and/or declare but not addhtml, 1: staff - mute but not addhtml, 0: regs, players, bots - the rest)

However the server-sent order shows bot higher than moderator in terms of jurisdiction order which in the userlist is obviously wrong.  By sorting the three auth types (0, 1, 2) within itself and compiling them in the order 2, 1 and 0, the order of ranks would compile itself properly that way, as bot is labelled as a type 0 (not bold or italics)

@Zarel